src/components/Link/Link.stories.mdx

```diff
@@ -0,0 +1,169 @@
+// import { boolean, text, select } from "@storybook/addon-knobs";
+
+<Meta
+  title="Link"
+  component={Link}
+  decorators={[withDesign]}
+  parameters={{
+    design: {
+      type: "figma",
+      url:
+        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=16115%3A407",
+    },
+  }}
+  argTypes={{
+    blockName: { table: { disable: true } },
+    modifiers: { table: { disable: true } },
+    children: { table: { disable: true } },
+    type: {
+      control: {
+        type: "select",
+      },
+    },
+  }}
+/>
+
+# Link
+
+<Description of={Link} />
+
+<Preview withToolbar>
+  <Story
+    name="Basic"
+    args={{
+      id: "nypl-link",
+      type: LinkTypes.Action,
+      href: "http://google.com",
+      className: "custom-class",
+      attributes: {
+        rel: "nofollow",
+        onClick: action("click"),
+      },
+    }}
+  >
+    {args => <Link {...args}>Link</Link>}
+  </Story>
+</Preview>
+
+<ArgsTable story="Basic" />
+
+## Links With Icons
+
+To pass in an `Icon`, make sure that the `Link` tag wraps the `Icon`. Use `type: "Action"` to apply appropriate styling to links with icons.
+
+<Preview>
+  <Story name="With Icons">
+    <div>
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        <Icon
+          name={"headset"}
+          modifiers={["left", "small"]}
+          decorative={true}
+        />
+        Headset Link
+      </Link>
+    </div>
+    <div>
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        <Icon name={"clock"} modifiers={["left", "small"]} decorative={true} />
+        Clock Link
+      </Link>
+    </div>
+    <div>
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        <Icon name={"check"} modifiers={["left", "small"]} decorative={true} />
+        Check Link
+      </Link>
+    </div>
+  </Story>
+</Preview>
+
+## Tag Structure
+
+`Link` can wrap an existing `<a>` tag or it can use the `href` prop to generate an `<a>` tag.
+
+<Preview>
+  <Story name="Tag Structure">
+    <div>
+      {"This is a "}
+      <Link type={LinkTypes.Action}>
+        <a href="#existing-anchor-tag">link</a>
+      </Link>
+      {" with the <a> tag as a child of the Link component. And this is a "}
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        link
+      </Link>
+      {
+        " where the Link component uses the `href` prop and has a string-only child."
+      }
+    </div>
+  </Story>
+</Preview>
+
+## Link with Other Routers
+
+### React Router
+
+The Design System's `Link` component should wrap around react-router's own `Link` component.
+
+<Preview>
+  <Story name="Link with Other Routers">
+    <Router>
+      <Link type={LinkTypes.Action}>
+        <ReactRouterLink to="#">
+          <Icon name="download" modifiers={["left", "small"]} decorative={true}></Icon>
+          Download
+        </ReactRouterLink>
+      </Link>
+    </Router>
+  </Story>
+</Preview>
+
+### NextJS Link
```

Comment:
@wluisi and @hokei 
I was unable to get the code from the `NextJS Link` section of the old `tsx` story to work here in the new `mdx` version.  For now, I am just showing the code in a code block.  If you have a chance, could you look at this and see if you can make it functional here?  Thanks.